digiroad2-viite/src/main/scala/fi/liikennevirasto/viite/dao/ProjectDAO.scala

```diff
@@ -933,6 +933,17 @@ object ProjectDAO {
     dateTime.map(dt => new Timestamp(dt.getMillis))
   }
 
+  def uniqueName(projectId: Long, projectName: String): Boolean = {
+    val query =
+      s"""
+         SELECT *
+         FROM project
+         WHERE UPPER(name)=UPPER('$projectName') and state<>7
```

Comment:
case (upper) shouldn't matter in SQL
To make query bit faster add "ROWNUM = 1" clause so if it finds one it stops searching

Reply:
UPPER actually seems to matter here. I'll add that "ROWNUM=1".